hypex/dataset/datagenerator.py

```diff
@@ -0,0 +1,293 @@
+        >>>
+        >>> print(sample_data.df)
+        >>>
+        >>>
+        >>> print(sample_data) # More details about attributes
+    """
+
+    def __init__(
+            self,
+            num_records: int = 5000,
+            num_features: int = 4,
+            num_info_cols: int = 2,
+            num_discrete_features: int = 1,
+            num_targets: int = 1,
+            is_treatment: bool = True,
+            is_binary_target: bool = False,
+            na_columns: Union[str, list, None] = None,
+            na_step: int = 0) -> None:
+        """
+        Initialize the DataGenerator object.
+
+        Args:
+            num_records:
+                Number of records in created dataset. Defaults to 5000
+            num_features:
+                Number of columns correlating with outcome. Defaults to 4
+            num_info_cols:
+                Number of columns with information about each record. Defaults to 2
+            is_treatment:
+                Number of treatments (info about 'treatment' 0 or 1). Defaults to 1
+            num_discrete_features:
+                Number of discrete columns correlating with outcome. Defaults to 1
+            num_targets:
+                Number of columns with target values. Defaults to 1
+            is_binary_target:
+                If outcome should be binary. Defaults to False
+            na_columns:
+                Column or list of columns with NA values. Defaults to None
+            na_step:
+                Number of period to make NaN (step of range). Defaults to 0
+        """
+        self.num_records = num_records
+        self.num_features = num_features
+        self.num_info_cols = num_info_cols
+        self.is_treatment = is_treatment
+        self.num_targets = num_targets
+        self.num_discrete_features = num_discrete_features
+        self.is_binary_target = is_binary_target
+        self.na_columns = na_columns
+        self.na_step = na_step
+        self.df: pd.DataFrame = pd.DataFrame()
+        self.treatment_name: Union[str, list, None] = None
+        self.target_names: list = []
+        self.info_col_names: Union[list, None] = None
+        self.features_names: list = []
+        self.ate: float = 0.0
+        self.sigma = 1.3
+        self._generate_df()
+
+    def _generate_df(self):
+        """
+        Generates the dataset and predict ate for it
+        """
+        m_with_dummy = self._generate_feature_cols()
+        info = self._generate_info_cols()
+        treatment = self._generate_treatment_cols()
+        data = self._generate_outcome_and_full_data(m_with_dummy, treatment)
+        self._set_column_names()
+        self.df = self._set_df(data, info)
+        self._set_nans()
+        self._process_ate()
+
+    def _process_ate(self):
+        pass
+
+    def _generate_feature_cols(self):
+        """
+        Generate columns with main features for causal process
+
+        Returns:
+            m_with_dummy:
```

Comment:
Suggestion: Maybe rename the variable that is returned here? m_with_dummy, but the function name is _generate_feature_cols. Might want to rename it so that it captures the essence with feature_cols